fgpyo/alignment/__init__.py

```diff
@@ -0,0 +1,232 @@
+from typing import Callable
+from typing import List
+
+from fgpyo.sam import Cigar
+from fgpyo.sam import CigarElement
+from fgpyo.sam import CigarOp
+
+
+class Alignment:
+    """
+    A general class to describe the alignment between two sequences or partial ranges thereof
+    """
+
+    def __init__(
+        self,
+        query: str,
+        target: str,
+        query_start: int,
+        target_start: int,
+        cigar: Cigar,
+        score: int,
+    ):
+        """
+        Args:
+            query: the query sequence
+            target: the target sequence
+            query_start: the 1-based position in the query sequence where the alignment begins
+            target_start: the 1-based position in the target sequence where the alignment begins
+            cigar: a Cigar object describing the alignment of the two sequences
```

Comment:
I tend not to use "class" or "object" in docstrings since they we know they are classes or objects (since that's what we're document)
```suggestion
            cigar: the Cigar describing the alignment of the two sequences
```